Minisat/mtl/Vec.h

```diff
@@ -106,9 +106,10 @@ void vec<T,_Size>::capacity(Size min_cap) {
     if (cap >= min_cap) return;
     Size add = max((min_cap - cap + 1) & ~1, ((cap >> 1) + 2) & ~1);   // NOTE: grow by approximately 3/2
     const Size size_max = std::numeric_limits<Size>::max();
-    if ( ((size_max <= std::numeric_limits<int>::max()) && (add > size_max - cap))
-    ||   (((data = (T*) REALLOC_UNKNOWN((void *)data, (cap += add) * sizeof(T),"Minisat::vec")) == NULL) && errno == ENOMEM) )
+    if ( ((size_max <= std::numeric_limits<int>::max()) && (add > size_max - cap)))
         throw OutOfMemoryException();
+
+    data = (T*) xrealloc((void *)data, (cap += add) * sizeof(T));
```

Comment:
I `xrealloc` expected to do the throwing corresponding to the second half of the original if-condition?

Reply:
Yes, it should.

Reply:
Actually, orig minisat does (https://github.com/niklasso/minisat/blame/master/minisat/mtl/Vec.h):

```
template<class T, class _Size>
void vec<T,_Size>::capacity(Size min_cap) {
    if (cap >= min_cap) return;
    Size add = max((min_cap - cap + 1) & ~1, ((cap >> 1) + 2) & ~1);   // NOTE: grow by approximately 3/2
    const Size size_max = std::numeric_limits<Size>::max();
    if ( ((size_max <= std::numeric_limits<int>::max()) && (add > size_max - cap))
    ||   (((data = (T*)::realloc(data, (cap += add) * sizeof(T))) == NULL) && errno == ENOMEM) )
        throw OutOfMemoryException();
 }
```

I wonder what `(T*)::realloc` even is. Some kind of "look in your namespace first"?

Reply:
I think that's "call the global realloc and cast to `T *`". I'm not sure if there is a difference between `realloc` and `std::realloc`.